watertap/unit_models/boron_removal.py

```diff
@@ -0,0 +1,1202 @@
+                        ] = approx_dens * value(
+                            self.control_volume.properties_in[t0].flow_mol_phase_comp[
+                                ind
+                            ]
+                            / sum(
+                                self.control_volume.properties_in[
+                                    t0
+                                ].flow_mol_phase_comp[ind[0], j]
+                                for j in self.config.property_package.component_list
+                            )
+                        )
+
+                # Check to see if 'conc_mass_phase_comp' is constructed
+                if self.control_volume.properties_out[t].is_property_constructed(
+                    "conc_mass_phase_comp"
+                ):
+                    approx_dens = pyunits.convert(
+                        1000 * pyunits.kg / pyunits.m**3,
+                        to_units=units_meta("mass") * units_meta("length") ** -3,
+                    )
+                    for ind in self.control_volume.properties_in[
+                        t
+                    ].conc_mass_phase_comp:
+                        self.control_volume.properties_out[t].conc_mass_phase_comp[
+                            ind
+                        ] = approx_dens * value(
+                            self.control_volume.properties_in[t0].flow_mol_phase_comp[
+                                ind
+                            ]
+                            * self.control_volume.properties_in[t0].mw_comp[ind[1]]
+                            / sum(
+                                self.control_volume.properties_in[
+                                    t0
+                                ].flow_mol_phase_comp[ind[0], j]
+                                * self.control_volume.properties_in[t0].mw_comp[j]
+                                for j in self.config.property_package.component_list
+                            )
+                        )
+
+                    for ind in self.control_volume.properties_in[
+                        t
+                    ].conc_mass_phase_comp:
+                        self.control_volume.properties_in[t].conc_mass_phase_comp[
+                            ind
+                        ] = approx_dens * value(
+                            self.control_volume.properties_in[t0].flow_mol_phase_comp[
+                                ind
+                            ]
+                            * self.control_volume.properties_in[t0].mw_comp[ind[1]]
+                            / sum(
+                                self.control_volume.properties_in[
+                                    t0
+                                ].flow_mol_phase_comp[ind[0], j]
+                                * self.control_volume.properties_in[t0].mw_comp[j]
+                                for j in self.config.property_package.component_list
+                            )
+                        )

```

Comment:
It might make sense to move this to a helper function so reduce the indentation two levels, making the code easier to read and reason about (e.g., am I still in this `if` block?)